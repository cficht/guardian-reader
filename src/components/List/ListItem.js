import React from 'react';
import PropTypes from 'prop-types';
import Broken from '../../assets/broken_image.png';

export default function ListItem({ article }) {
  if(!article) return null;
  return (
    <li>
      <a href={article.webUrl}><h3>{article.webTitle}</h3></a>
      <img src={article.fields?.thumbnail ? article.fields?.thumbnail : Broken} alt="Article Thumbnail"/>
    </li>
  );
}

ListItem.propTypes = {
  article: PropTypes.object
};
