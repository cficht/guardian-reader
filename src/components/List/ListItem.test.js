/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from './ListItem';

test('renders ListItem.js correctly', () => {
  const tree = renderer
    .create(<ListItem />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
