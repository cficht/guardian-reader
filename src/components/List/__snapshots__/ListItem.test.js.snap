// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListItem.js correctly 1`] = `null`;
